### ISSUE TITLE
ai: caro ai --once injects placeholder command with misleading confidence=0.85 risk=Safe when no backend available

### ISSUE THREAD

**Issue by wildcard** [wildcard/caro#1424]
`[agent]`

**Agent:** Claude Code (`claude-sonnet-4-6`) — caro-qa-agent

---

## Problem

`caro ai --once` exits 0 and emits the following stdout when no model is downloaded and no remote backend is configured:

```
# caro-ai: session 1 confidence=0.85 risk=Safe
echo 'Please clarify your request'
```

The `ai` help text advertises that "Stdout is the generated command only, so shell widgets can inject it straight into the prompt buffer." A shell integration that consumes this output will silently inject `echo 'Please clarify your request'` into the user's readline buffer — requiring manual deletion and giving no hint that the backend was unavailable. The `confidence=0.85` and `risk=Safe` labels are hardcoded defaults (`src/backends/embedded/embedded_backend.rs:511`), not values derived from actual model output, so shell widgets have no signal that the result is a fallback.

The same behaviour is observed on `caro -p "..." --dry-run` (Slot A smoke), which also exits 0 with the placeholder when no model is present.

## Reproduction

```bash
# In a fresh environment with no model downloaded and no Ollama installed:
cargo build --release --features embedded-cpu
echo "list current directory" | ./target/release/caro ai --once
# Stdout:
# # caro-ai: session 1 confidence=0.85 risk=Safe
# echo 'Please clarify your request'
# Exit: 0

echo "show disk usage" | ./target/release/caro ai --once
# Same output — confidence and command are invariant across prompts

echo "kill process 1234" | ./target/release/caro ai --once
# Same output — risk=Safe even for a potentially destructive prompt
```

## Expected vs Actual

**Expected:** When no backend is available, `caro ai --once` should exit non-zero (e.g. exit 1) with a diagnostic on stderr and no command on stdout, so shell widgets know to abort injection.

**Actual:** Exits 0 with `confidence=0.85 risk=Safe` and `echo 'Please clarify your request'` on stdout regardless of prompt content or backend availability.

## Environment

- caro version: `caro 1.5.0 (be07b22 2026-07-18)`
- OS: `Linux 6.18.44-fc-v21 x86_64` (remote sandbox)
- Test date: 2026-08-24
- Backend: embedded (no model downloaded; Ollama not installed)

## Investigation

- `src/backends/embedded/embedded_backend.rs:511` — `confidence_score: 0.85` is a hardcoded constant, not derived from inference quality.
- `src/backends/embedded/embedded_backend.rs::is_available()` — returns `true` unconditionally even before model download, so the agent loop selects the embedded backend and attempts generation.
- `src/backends/embedded/cpu.rs:63-64` — placeholder fallback `echo 'Please clarify your request'` is hardcoded for unrecognised or "delete/rm" prompts.
- `src/main.rs:1206` — the `# caro-ai:` header line formats `confidence` and `risk` from the `GeneratedCommand` struct; both reflect the hardcoded defaults rather than real signal.
- The `WARN caro::agent: Timeout approaching, skipping refinement` log line confirms the refinement step was skipped due to timeout, yet exit code remains 0.

## Fix Direction

1. `is_available()` in the embedded backend should return `false` (or check `~/.cache/caro/models/` for a downloaded model) rather than unconditionally `true` before model download completes.
2. When every configured backend is unavailable, the `ai --once` path should exit non-zero and emit a clear error on stderr only, with nothing on stdout, so shell widgets observe a failure exit code.
3. `confidence_score: 0.85` should either be computed from model perplexity/logits or removed in favour of a backend-availability flag that downstream code can inspect.

## Severity

P1 — Feature works (binary doesn't crash) but not as documented: the advertised shell-widget stdout contract (`"the generated command only"`) is violated. A user who wires `caro ai --once` into their shell keybinding receives a garbage command silently injected into their readline buffer whenever the model is absent.

---

<details>
<summary>Prompt used to generate this comment</summary>

```
caro QA agent — daily rotation slot A+B+C, scheduled remote run 2026-08-24
```

</details>
